### PR TITLE
mattdrayer/from-email-default: Allow null field values

### PR DESCRIPTION
**Pull request by mattdrayer**
@clintonb @mjfrey -- including a default value for the new SiteConfiguration.from_email model field.

### CODE REVIEW

File: ecommerce/core/migrations/0015_siteconfiguration_from_email.py [hunk 1]
```diff
@@ -14,7 +14,6 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='siteconfiguration',
             name='from_email',
-            field=models.CharField(default='oscar@example.com', help_text='Address from which emails are sent.', max_length=255, verbose_name='From email'),
-            preserve_default=False,
+            field=models.CharField(default=b'oscar@example.com', help_text='Address from which emails are sent.', max_length=255, verbose_name='From email'),
```
Comment (mattdrayer) [inline ecommerce/core/migrations/0015_siteconfiguration_from_email.py]: Hmm, this isn't doing what I would expect -- is there a way to make this a dynamic value?  Can I modify the actual migration file to pull settings.OSCAR_FROM_EMAIL?

File: ecommerce/core/models.py [hunk 2]
```diff
@@ -65,7 +65,8 @@ class SiteConfiguration(models.Model):
         help_text=_('Address from which emails are sent.'),
         max_length=255,
         null=False,
```
Comment (clintonb) [inline ecommerce/core/models.py]: Skip the default and make the field nullable.

Comment (mattdrayer) [inline ecommerce/core/models.py]: Okay, I've updated the model to allow for null values and regenerated the migration file.

**Comment by clintonb** [comment]
:+1:

File: ecommerce/core/migrations/0015_siteconfiguration_from_email.py [hunk 3]
```diff
@@ -14,7 +14,6 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='siteconfiguration',
             name='from_email',
-            field=models.CharField(default='oscar@example.com', help_text='Address from which emails are sent.', max_length=255, verbose_name='From email'),
-            preserve_default=False,
+            field=models.CharField(help_text='Address from which emails are sent.', max_length=255, null=True, verbose_name='From email', blank=True),
```
Comment (mjfrey) [inline ecommerce/core/migrations/0015_siteconfiguration_from_email.py]: Do you need a new migration?

**Pull request merged**